shared/src/main/scala/io/lemonlabs/uri/Uri.scala

```diff
@@ -129,6 +129,7 @@ sealed trait Url extends Uri {
   type SelfWithScheme <: Url
   type SelfWithAuthority <: UrlWithAuthority
 
+  def authorityOption: Option[Authority] = None
```

Comment:
Can we make this abstract (`def authorityOption: Option[Authority]`) and force subclasses to implement. Then if we add a new subclass of Url in the future we won't forget to implement

Reply:
Sure!